schema/tmpBuilders.go

```diff
@@ -149,6 +149,93 @@ func SpawnEnum(name TypeName, members []string, repr EnumRepresentation) *TypeEn
 	return &TypeEnum{typeBase{name, nil}, members, repr}
 }
 
+// Utility function adding default basic types to schema type system
+func SpawnDefaultBasicTypes(ts *TypeSystem) {
+	ts.Accumulate(SpawnBool("Bool"))
+	ts.Accumulate(SpawnInt("Int"))
+	ts.Accumulate(SpawnFloat("Float"))
+	ts.Accumulate(SpawnString("String"))
+	ts.Accumulate(SpawnBytes("Bytes"))
+
+	ts.Accumulate(SpawnAny("Any"))
+
+	ts.Accumulate(SpawnMap("Map", "String", "Any", false))
+	ts.Accumulate(SpawnList("List", "Any", false))
+
+	// Should be &Any, really.
+	ts.Accumulate(SpawnLink("Link"))
+
+	// TODO: schema package lacks support?
+	// ts.Accumulate(schema.SpawnUnit("Null", NullRepr))
+}
+
+// Clone creates a copy of a type that is not in the original's universe (so it can be used elsewhere safely)
+func Clone(typ Type) Type {
+	switch kindedType := typ.(type) {
+	case *TypeBool:
+		return SpawnBool(kindedType.Name())
+	case *TypeString:
+		return SpawnString(kindedType.Name())
+	case *TypeBytes:
+		return SpawnBytes(kindedType.Name())
+	case *TypeInt:
+		return SpawnInt(kindedType.Name())
+	case *TypeFloat:
+		return SpawnFloat(kindedType.Name())
+	case *TypeAny:
+		return SpawnAny(kindedType.Name())
+	case *TypeMap:
+		return SpawnMap(kindedType.Name(),
+			kindedType.KeyType().Name(),
+			kindedType.ValueType().Name(),
+			kindedType.ValueIsNullable())
+	case *TypeList:
+		return SpawnList(kindedType.Name(), kindedType.ValueType().Name(), kindedType.ValueIsNullable())
```

Comment:
hah, we don't store representation strategy for these even though we have space for it in the dmt, TODO at some point I guess